experimental/net/net.go

```diff
@@ -0,0 +1,57 @@
+package net
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"strings"
+
+	"github.com/tetratelabs/wazero/internal/net"
+)
+
+// NetConfig configures the host to set up host:port listeners
+// that the embedding host will allow the wasm guest to access.
+//
+// These will result in preopened socket file-descriptors at run-time.
+type NetConfig interface {
+	// WithTCPListener configures the host to set up the given host:port listener.
+	WithTCPListener(host string, port uint16) NetConfig
+	// WithTCPListenerFromString parses a string of the form "host:port"
+	// into a host:port pair.
+	WithTCPListenerFromString(conn string) (NetConfig, error)
```

Comment:
in a commit, I'll remove this as there's no need to add encoding logic at this level of abstraction.